data-dot-all/dataall
# 922 -  Replace IAM inline policies by configurable Managed Policies for folder and bucket sharing

### Feature or Bugfix
- Feature


### Detail
- For each consumption role and group role, invited to the env 1 managed policy is created
- User can choose, if the consumption role is data.all managed or not
- If data.all managed, the created policy is automatically attached to role
- without this policy attached user can not create a share for this consumption role
- policy attachment and managed options are displayed in Env->Teams
- bucket and accesspoint shares are managed through this managed policies

### Relates
- [922](https://github.com/data-dot-all/dataall/issues/922)

### SecurityN/A

How to test:
See backwards compatibility and local testing comments below.
 
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.


@dlpzx Update:
- ⚠️ When upgrading customers need to update their environment stacks to update the pivot role if created through CDK. If the pivot role is manually created, they need to update its permissions manually with the latest version
- ⚠️ For backwards compatibility: for all existing roles both created and imported by data.all. The share-policy will be created in any of the next 3 scenarios:
a) if a new share request is created
b) if new items are added to an existing share request 
c) if items are revoked from an existing share request 

More details in the comments
